### ISSUE TITLE
Could not load source file "../src/Typesense.ts" in source map of "node_modules/typesense/lib/Typesense.js"

### ISSUE THREAD

**Issue by jasonbosco** [typesense/typesense-js#86]
## Description

Running `parcel build` shows these warnings:

```
00:21:36.587 | $ parcel build index.html --public-url https://airbnb-geosearch.typesense.org
-- | --
00:21:45.993 | ⚠️  Could not load source file "../src/Typesense.ts" in source map of "node_modules/typesense/lib/Typesense.js".
00:21:46.657 | ⚠️  Could not load source file "../src/Typesense.ts" in source map of "node_modules/typesense/lib/Typesense.js".
00:21:53.342 | ⚠️  Could not load source file "../../src/Typesense/Client.ts" in source map of "node_modules/typesense/lib/Typesense/Client.js".
00:21:53.347 | ⚠️  Could not load source file "../../src/Typesense/SearchClient.ts" in source map of "node_modules/typesense/lib/Typesense/SearchClient.js".
00:21:56.321 | ⚠️  Could not load source file "../../src/Typesense/Aliases.ts" in source map of "node_modules/typesense/lib/Typesense/Aliases.js".
00:21:56.322 | ⚠️  Could not load source file "../../src/Typesense/Alias.ts" in source map of "node_modules/typesense/lib/Typesense/Alias.js".
00:21:56.322 | ⚠️  Could not load source file "../../src/Typesense/Keys.ts" in source map of "node_modules/typesense/lib/Typesense/Keys.js".
00:21:56.323 | ⚠️  Could not load source file "../../src/Typesense/Key.ts" in source map of "node_modules/typesense/lib/Typesense/Key.js".
00:21:56.324 | ⚠️  Could not load source file "../../src/Typesense/Debug.ts" in source map of "node_modules/typesense/lib/Typesense/Debug.js".
00:21:56.338 | ⚠️  Could not load source file "../../src/Typesense/Configuration.ts" in source map of "node_modules/typesense/lib/Typesense/Configuration.js".
```

## Steps to reproduce

1. Clone https://github.com/typesense/showcase-airbnb-geosearch
2. `yarn install`
3. `parcel build`

## Expected Behavior

No warnings are logged

## Actual Behavior

The warnings above are logged

## Metadata

**Typsense.js Version**: 1.0.3-2

**Comment by jasonbosco** [comment]
@dcantu476 @AadamZ5

Any thoughts on this? The `.ts` src files are currently not published to npm, which I'm guessing is why these warnings show up. Is it standard practice to publish the `src` directory to npm for Typescript packages?

**Comment by dcantu476** [comment]
I'll take a peek at it when I can, but an extra set of eyes is welcome as I'm unfamiliar with parcel. It appears it's trying to package source maps as part of the build pipeline. 

Professionally speaking, you should only publish artifacts such as the dist or lib folder with the *.d.ts files (type definitions) and compiled *.js files, **not src files.** On a practical level, though, you'll find plenty of packages with src folders published.

**Comment by AadamZ5** [comment]
I think @dcantu476 is definitely right. Looks like this library is publishing `*.js.map` files, which tell compilers where to find source code. In a development build, this is useful. However, for production and distribution, these `*.js.map` files shouldn't be included since no source code is distributed with it. End users should be debugging this library inside of their own projects.

You can see inside `lib/Typesense.js.map` what it contians 
```JSON
{
  "version":3,
  "file":"Typesense.js",
  "sourceRoot":"",
  "sources":["../src/Typesense.ts"],
  "names":[],
  "mappings":"...<omitted>..."
}
```

Any compiler/interpreter/debugger/whatever reading this file will look for the path(s) at the `"sources"` key, which is `["../src/Typesense.ts"]`, which doesn't exist in the released and published version.

I think for release, this library needs to be configured with different options so that the TS compiler doesn't generate `*.js.map` files. (`*.d.ts` files should stay). This could be done by [extending TS config files](https://www.typescriptlang.org/tsconfig#extends). During a release build, point the `tsc` compiler to the release TS configuration file, that extends and overrides the default/development config file. 

A production/release TS config file (maybe named "tsconfig.prod.json") might look like this:
```JSON
{
  "extends": "./tsconfig",
  "compilerOptions": {
    "sourceMap": false
  }
}
```

Then you'd have to build using this new config before publishing a new version to the package repository. Be careful, I don't think that `tsc` cleans/deletes the existing `lib` folder. 

Hope that explanation wasn't confusing 😅

**Comment by jasonbosco** [comment]
Thank you for the detailed explanation Damian and Aadam! 

IIRC, I might have explicitly turned on source maps several years ago because I found it helpful to debug and get to the bottom of bugs quickly, when I used the library myself in other projects. The source maps are generated as separate files though, so they don't end up getting including in users' production builds (or at least that was my understanding of how it works). Is it not standard for libraries to bundle source maps (as separate files) in NPM packages this way? 

If not, if users of the library wanted to report bugs they see, wouldn't it be hard to get stack traces without source maps?
